region/lowernorfair/east.json

```diff
@@ -537,39 +552,114 @@
               "requires": [
                 {"or": [
                   {"and": [
-                    "canKillYellowSpacePirates",
+                    "heatProof",
+                    {"or": [
+                      "canKillYellowSpacePirates",                          
+                      {"canComeInCharged": {
+                        "fromNode": 2,
+                        "framesRemaining": 120
+                      }}
+                    ]}
+                  ]},
+                  {"and": [
+                    "canHeatRun",
                     {"or": [
                       {"and": [
                         "ScrewAttack",
-                        "SpaceJump"
+                        {"cost": [
+                          {"heatFrames": 250}
+                        ]}
+                      ]},
+                      {"and": [
+                        "Charge",
+                        "Plasma",
+                        {"cost": [
+                          {"heatFrames": 600}
+                        ]}
                       ]},
-                      "canIBJ",
+                      {"cost": [
+                        {"heatFrames": 450},
+                        {"ammo": [{
+                          "type": "Super",
+                          "count": 9
+                        }]}
+                      ]},
+                      {"and": [
+                        "Charge",
+                        "Ice",
+                        "Spazer",
+                        "Wave",
+                        {"cost": [
+                          {"heatFrames": 1150}
+                        ]}
+                      ]},
+                      {"canComeInCharged": {
+                        "fromNode": 2,
+                        "framesRemaining": 120
+                      }}
+                    ]}
+                  ]}
+                ]},
+                {"or": [
+                  {"and": [
+                    {"canComeInCharged": {
```

Comment:
I think the shinespark needs to be entirely separate from the "kill pirates vs climb up" split, because otherwise combinations that use a shinespark for only one of the two are in logic even though they don't make sense.

Reply:
Would have to add extra heatProof / canHeatRun into it but it's more clean than two canComeInCharged objects for sure.